ts/preload.ts

```diff
@@ -788,25 +858,41 @@ export class ThemeUtils {
   }
 }
 
+new Nody();
 new ThemeUtils();
 new GreeterConfig();
 new Greeter();
 
 window._ready_event = new Event("GreeterReady");
 
-window.addEventListener("DOMContentLoaded", () => {
-  setTimeout(() => {
-    window.dispatchEvent(globalThis._ready_event);
-  }, 2);
+const domLoaded = new Promise<void>(resolve => {
+  window.addEventListener("DOMContentLoaded", () => {
+    setTimeout(() => {
+      resolve();
+    }, 2);
+  });
 });
 
+/**
+ * Promise that fires when all initialization has completed,
+ * and the theme can start (i.e. _ready_event can be sent)
+ */
+const readyPromise = Promise.all([
+  domLoaded,
+  globalThis.nody_greeter.whenReady()
+]);
+
+readyPromise.then(() => window.dispatchEvent(globalThis._ready_event))
+
+export declare const nody_greeter: Nody;
 export declare const lightdm: Greeter;
 export declare const greeter_config: GreeterConfig;
 export declare const theme_utils: ThemeUtils;
 export declare const _ready_event: Event;
 
 declare global {
   interface Window {
+    nody: Nody | undefined;
```

Comment:
```suggestion
    nody_greeter: Nody | undefined;
```

The constants inside the Window interface should correspond to their existing variable. Thus, the `nody-greeter-types` package could suggest `nody_greeter` (which does exist), instead of `nody` (which doesn't).

Reply:
ah whoops not sure how i missed this, think i was relying on TypeScript being in strict mode, and decided to rename the variable half way through! :sweat_smile: Nice catch!